apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -3975,27 +3975,26 @@ fn handle_requires(
         let parent_type = new_node.parent_type.clone();
         for created_node_id in &new_created_nodes {
             let created_node = dependency_graph.node_weight(*created_node_id)?;
+            // Usually, computing the path of our new group into the created groups
+            // is not entirely trivial, but there is at least the relatively common
+            // case where the 2 groups we look at have:
+            // 1) the same `mergeAt`, and
+            // 2) the same parentType; in that case, we can basically infer those 2
+            //    groups apply at the same "place" and so the "path in parent" is
+            //    empty. TODO: it should probably be possible to generalize this by
+            //    checking the `mergeAt` plus analyzing the selection but that
+            //    warrants some reflection...
```

Comment:
please make a task for this in Jira so we don't lose this thought in a comment.

Reply:
this is old Sylvain's note ported over